### ISSUE TITLE
Add explicit file mode to h5py.File() call

### ISSUE THREAD

**Issue by Caseyb87** [spacepy/spacepy#444]
The file mode should be set for any h5py.File() calls to prevent a warning message being given to user.

The call is used when reading in OMNI files and any dependent functions (e.g. getMPstandoff). 

### Offending lines in omni.py
```
274: with h5.File(omni2fln) as hfile:
376: with h5.File(infile[dbase]) as hfile:
```
Note, is fixed in
```
248: with h5.File(fln, 'r') as hfile:
```

### Error message/Traceback:

```
C:\Users\username\Anaconda3\lib\site-packages\spacepy\omni.py:376: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.
  with h5.File(infile[dbase]) as hfile:
```

### OS, Python version, and dependency version information:

```
Windows-10-10.0.19041-SP0
sys.version_info(major=3, minor=8, micro=3, releaselevel='final', serial=0)
numpy=1.19.1
scipy=1.5.0
matplotlib=3.3.1
```

**Comment by balarsen** [comment]
Thanks for this, I suspected it was going to come.

**Comment by balarsen** [comment]
This is actually already fixed. I will find the PR for reference.

**Comment by balarsen** [comment]
This is a duplicate of #322, was closed in #326.
